lib/ruby_indexer/lib/ruby_indexer/configuration.rb

```diff
@@ -174,5 +170,39 @@ def apply_config(config)
       @included_patterns.concat(config["included_patterns"]) if config["included_patterns"]
       @excluded_magic_comments.concat(config["excluded_magic_comments"]) if config["excluded_magic_comments"]
     end
+
+    sig { returns(T::Array[String]) }
+    def initial_excluded_gems
+      excluded, others = Bundler.definition.dependencies.partition do |dependency|
+        dependency.groups == [:development]
+      end
+
+      # When working on a gem, we need to make sure that its gemspec dependencies can't be excluded. This is necessary
+      # because Bundler doesn't assign groups to gemspec dependencies
+      this_gem = Bundler.definition.dependencies.find { |d| d.to_spec.full_gem_path == Dir.pwd }
+      others.concat(this_gem.to_spec.dependencies) if this_gem
```

Comment:
I'm confused by the `this_gem` logic, could you elaborate a bit. I feel like `Dir.pwd` will always return `ruby-lsp` so we will add its dependencies to `others`?

Reply:
Yeah, I couldn't think of a better name for it.

Basically, if you're working on a gem, we need to prevent the gem's gemspec dependencies from being excluded accidentally.

For example, in the Ruby LSP we have `sorbet-static-and-runtime` in the Gemfile, but `sorbet-runtime` as a gemspec dependency. Because `sorbet-runtime` is a transitive dependency of `sorbet-static-and-runtime` and that gem is in the development group, we would accidentally exclude a real dependency.

Reply:
Ah I see. Thanks, I was mistaken `Dir.pwd` does return directory of the main application.